bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/thing/ThingResource.java

```diff
@@ -515,6 +520,65 @@ public Response getConfigStatus(@HeaderParam(HttpHeaders.ACCEPT_LANGUAGE) String
         return Response.ok().entity(Collections.EMPTY_SET).build();
     }
 
+    @PUT
+    @Path("/{thingUID}/firmware/{firmwareVersion}")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "Update thing firmware.")
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "OK"),
+            @ApiResponse(code = 400, message = "Unkown thing or firmware") })
+    public Response updateFirmware(
+            @HeaderParam(HttpHeaders.ACCEPT_LANGUAGE) @ApiParam(value = "language") String language,
+            @PathParam("thingUID") @ApiParam(value = "thing") String thingUID,
+            @PathParam("firmwareVersion") @ApiParam(value = "version") String firmwareVersion) throws IOException {
+        Thing thing = thingRegistry.get(new ThingUID(thingUID));
+        if (thing == null) {
+            return JSONResponse.createResponse(Status.BAD_REQUEST, null, "Unknown thingUID");
```

Comment:
agreed. and there is already `getThingNotFoundResponse(thingUID)` in place for this. fixed.